src/components/metadataFormControls/ModelSingleSelect/ModelSingleSelect.tsx

```diff
@@ -84,6 +87,16 @@ export const ModelSingleSelect = <
         return transform ? transform(allDataMap) : allDataMap
     }, [allDataMap, transform])
 
+    useEffect(() => {
+        // fetch until we have resolvedAvailable length equal to the desired page size (or until we reach the end)
+        if (
+            resolvedAvailable.length < (queryObject.params.pageSize ?? 10) &&
+            queryResult.hasNextPage
+        ) {
+            queryResult.fetchNextPage()
+        }
+    }, [resolvedAvailable])
```

Comment:
Could we do `resolvedAvailable.length` as the dependency? `transform`-function may not be stable, and thus `resolvedAvailable` would not be - and this could potentially run in a loop?

Reply:
Right. Thanks 🙏. This I forgot to clean up.

Adding proper dependencies and not just using resolvedAvailable does however introduce the bug that I mentioned to you yesterday (that I thought we would have): once you get the first 10, and now go to reach the end and need to get the next ones, you can run into issues (e.g. your next refetch gives you no new resolvedAvailable, so you get stuck with the infinite loader again).

I've refined the logic in this commit (https://github.com/dhis2/maintenance-app-beta/pull/532/commits/7ca929ec8501c208ed24954272bb46e9c9188e22). So now you should get in blocks of 10 whenever you get to the end.